File: testdata/Mock.java
// Test fixtures for hollowcheck - mock data patterns.
package testdata;

import java.util.Arrays;
import java.util.List;

/**
 * Represents a user with fake data.
 */
class MockUser {
    String id;
    String email;
    String name;

    MockUser(String id, String email, String name) {
        this.id = id;
        this.email = email;
        this.name = name;
    }
}

/**
 * Has placeholder values.
 */
class MockConfig {
    String apiKey;
    String endpoint;
    String password;

    MockConfig(String apiKey, String endpoint, String password) {
        this.apiKey = apiKey;
        this.endpoint = endpoint;
        this.password = password;
    }
}

public class Mock {
    public static MockUser getTestUser() {
        return new MockUser("12345", "test@example.com", "foo");
    }

    public static MockUser getAnotherUser() {
        return new MockUser("00000", "user@example.com", "bar");
    }

    public static MockConfig getMockConfig() {
        return new MockConfig(
            "asdf1234",
            "https://api.example.com/v1",
            "changeme"
        );
    }

    public static String getDescription() {
        return "lorem ipsum dolor sit amet";
    }

    public static String getPhoneNumber() {
        return "xxx-xxx-xxxx";
    }

    public static List<String> sequentialIds() {
        return Arrays.asList("11111", "22222", "33333");
    }
}

File: testdata/Mock.kt
// Test fixtures for hollowcheck - mock data patterns.
package testdata

/**
 * Represents a user with fake data.
 */
data class MockUser(
    val id: String,
    val email: String,
    val name: String
)

/**
 * Has placeholder values.
 */
data class MockConfig(
    val apiKey: String,
    val endpoint: String,
    val password: String
)

fun getTestUser(): MockUser {
    return MockUser(
        id = "12345",
        email = "test@example.com",
        name = "foo"
    )
}

fun getAnotherUser(): MockUser {
    return MockUser(
        id = "00000",
        email = "user@example.com",
        name = "bar"
    )
}

fun getMockConfig(): MockConfig {
    return MockConfig(
        apiKey = "asdf1234",
        endpoint = "https://api.example.com/v1",
        password = "changeme"
    )
}

fun getDescription(): String {
    return "lorem ipsum dolor sit amet"
}

fun getPhoneNumber(): String {
    return "xxx-xxx-xxxx"
}

fun sequentialIds(): List<String> {
    return listOf("11111", "22222", "33333")
}
